lib/src/vector_math/matrix4.dart

```diff
@@ -638,20 +638,29 @@ class Matrix4 {
   Matrix4 operator -() => clone()..negate();
 
   /// Returns a new vector or matrix by multiplying this with [arg].
+  ///
+  /// [arg] should be a [double] (to scale), [Vector4] (to transform), [Vector3]
+  /// (to transform), or [Matrix4] (to multiply).
+  ///
+  /// If you know the argument type in a call site, prefer [scaled],
+  /// [transformed], [transformed3], or [multiplied] for performance.
+  @pragma('wasm:prefer-inline')
+  @pragma('vm:prefer-inline')
+  @pragma('dart2js:prefer-inline')
   dynamic operator *(dynamic arg) {
+    final dynamic value;
     if (arg is double) {
-      return scaled(arg);
-    }
-    if (arg is Vector4) {
-      return transformed(arg);
-    }
-    if (arg is Vector3) {
-      return transformed3(arg);
-    }
-    if (arg is Matrix4) {
-      return multiplied(arg);
+      value = scaled(arg);
+    } else if (arg is Vector4) {
+      value = transformed(arg);
+    } else if (arg is Vector3) {
+      value = transformed3(arg);
+    } else if (arg is Matrix4) {
+      value = multiplied(arg);
+    } else {
+      throw ArgumentError(arg);
     }
-    throw ArgumentError(arg);
+    return value;
```

Comment:
I don't think it is worthwhile making the `dynamic` versions faster like this.
The user has to do a dynamic call or cast to use the result.
I'd rather they were deprecated with proper versioning.

Reply:
But it's free performance, why not make it faster?

(It's free because I assume all calls sites will directly call it, so it'll always be inlined, and the type tests will always disappear as the argument types will also known at the call sites. So no binary size increase.)

I added deprecations.